crates/dids/src/method/dht/mod.rs

```diff
@@ -0,0 +1,148 @@
+mod bep44;
+mod convert;
+mod registered_types;
+
+use std::sync::Arc;
+
+use crypto::Curve;
+use keys::key_manager::KeyManager;
+use reqwest::blocking::Client;
+
+use crate::{
+    bearer::BearerDid,
+    document::{Document, Service, VerificationMethod},
+    identifier::Identifier,
+    resolver::{ResolutionMetadata, ResolutionResult},
+};
+
+use self::{
+    bep44::encode_bep44_message, convert::document_packet::document_to_packet,
+    registered_types::RegisteredDidType,
+};
+
+use super::{Method, MethodError};
+
+const DEFAULT_TTL: u32 = 7200; // seconds
+const DEFAULT_GATEWAY_URL: &str = "https://diddht.tbddev.org";
+const DID_DHT_SPEC_VERSION: u32 = 0;
+const NAME: &'static str = "dht";
+
+/// Concrete implementation for a did:dht DID
+pub struct DidDht;
+
+/// Options that can be used to create a did:dht DID
+pub struct DidDhtCreateOptions {
+    pub publish: bool,
+    // TODO: gatewayUri - default 'https://diddht.tbddev.org'
+    pub also_known_as: Option<Vec<String>>,
+    pub controller: Option<Vec<String>>,
+    pub service: Option<Vec<Service>>,
+    pub registered_type: Option<Vec<RegisteredDidType>>,
+    pub verification_methods: Option<Vec<VerificationMethod>>,
+}
+
+impl Method<DidDhtCreateOptions> for DidDht {
+    const NAME: &'static str = "dht";
+
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidDhtCreateOptions,
+    ) -> Result<BearerDid, MethodError> {
+        // Generate private and public keypair
+        let key_alias = key_manager.generate_private_key(
+            Curve::Ed25519,
+            Some("0".to_string()), // PR review: i copied this from jwk.rs. do we want this to be None?
+        )?;
+        let public_key = key_manager.get_public_key(&key_alias)?;
+
+        // Create did uri
+        let public_jwk = public_key.jwk()?;
+        let jwk_string = serde_json::to_string(public_jwk.as_ref()).map_err(|_| {
```

Comment:
either way works, either are fine with me

```suggestion
        let jwk_string = serde_json::to_string(&public_jwk).map_err(|_| {
```

Reply:
I kinda feel like we'd be wiser to expand our `MethodError` enum set to include more specific errors, and then write an `impl From` ([like we do here](https://github.com/TBD54566975/web5-rs/blob/b847389b5ec619583b643ca59a2d54ac87f11e65/crates/jws/src/lib.rs#L33)) and then we wouldn't have to call `.map_err`

I have nothing against `map_err` so that doesn't really matter either way, but I do think it's good practice to have error enums which are semantically relevant rather than relying on string messages... I'm going to open a ticket for this. no need to address right now unless you feel choose to.

Reply:
https://github.com/TBD54566975/web5-rs/issues/191

Reply:
Edited the original comment, as that suggestion is incorrect. Apologies.

Reply:
FYI, we should do this anyways https://github.com/TBD54566975/web5-rs/pull/199